### ISSUE TITLE
Support `bucketingprocess.prebucket_table(column)` for columns that are not prebucketed

### ISSUE THREAD

**Issue by timvink** [timvink/skorecard#60]
In the example below `EDUCATION` and `MARRIAGE` are not prebucketed but are bucketed. That's fine because they don't have a lot of unique values. This is however a problem when using the dash app, because the app shows the prebucket table and bucketing table side-by-side (see screenshot below).

```python
    from skorecard import datasets
    from skorecard.bucketers import DecisionTreeBucketer, OptimalBucketer
    from skorecard.pipeline import BucketingProcess
    from sklearn.pipeline import make_pipeline
    from sklearn.preprocessing import OneHotEncoder
    from sklearn.linear_model import LogisticRegression

    df = datasets.load_uci_credit_card(as_frame=True)
    y = df["default"]
    X = df.drop(columns=["default"])

    num_cols = ["LIMIT_BAL", "BILL_AMT1"]
    cat_cols = ["EDUCATION", "MARRIAGE"]

    bucketing_process = BucketingProcess()
    bucketing_process.register_prebucketing_pipeline(
                                DecisionTreeBucketer(variables=num_cols, max_n_bins=100, min_bin_size=0.05),
    )
    bucketing_process.register_bucketing_pipeline(
            OptimalBucketer(variables=num_cols, max_n_bins=10, min_bin_size=0.05),
            OptimalBucketer(variables=cat_cols, max_n_bins=10, min_bin_size=0.05),
    )

    pipeline = make_pipeline(
        bucketing_process,
        OneHotEncoder(),
        LogisticRegression()
    )
```

![image](https://user-images.githubusercontent.com/5570380/114714121-18818880-9d32-11eb-8870-4e0d33eee574.png)

We should do 2 things:

- In `bucketingprocess.fit()`, make sure all columns that are bucketed have a most X unique values (set default of 100 in __init__). 
- In `bucketingprocess.prebucket_table()` support using the unique values as bucket index and label, if number of unique values are below threshold.

**Comment by sbjelogr** [comment]
In principle you should pre-bucket the categoricals as well with the OrdinalCategoricalBucketer
```python
OrdinalCategoricalBucketer(variables=cat_col, tol=0) # by setting tol=0, no grouping of categories happens
```
This could be done even automatically in the code (though you need to know what the feature is categorical), or make sure that the user explicitly takes care of this?

**Comment by timvink** [comment]
If the user does not prebucket a column (by not applying a bucketer on the column in `.register_prebucketing_pipeline()`, the best thing is to assume the user wants to bucket directly on the unique values instead of new prebuckets.

We could require the user to explicitly add a `AsIsBucketer()` for columns that are not prebucketed, but I think we discussed this before and decided it was a bit of unnecessary cognitive overhead for the user. 

We do need to check that all columns have <X unique values after prebucketing, and otherwise require the user to apply a prebucketer on the column.
